firedrake/tsfc_interface.py

```diff
@@ -222,8 +225,10 @@ def tuplify(params):
     for idx, f in iterable:
         f = _real_mangle(f)
         # Map local coefficient numbers (as seen inside the
-        # compiler) to the global coefficient numbers
-        number_map = dict((n, coefficient_numbers[c])
+        # compiler) to the split global coefficient numbers
+        number_map = dict((n, (coefficient_numbers[c], tuple(range(len(c.split())))))
+                          if isinstance(c, Function) or isinstance(c, Constant)
```

Comment:
When is a coefficient not a function or constant?

Reply:
When it is a ufl `Coefficient`. The problem was that those coefficient do not have a `split` function.